### PR TITLE
Implicit trust

### PR DESCRIPTION
**Pull request by mcr**
replace pull request #60 with one we can edit.
This attempts to establish the architectural basis for implicit trust as a result of links.

### CODE REVIEW

File: draft-ietf-rats-architecture.md [hunk 1]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+Implicit trust can also be tied to the communications link over which
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:700]:
```suggestion
Trust can also be tied to the communications link over which
```

File: draft-ietf-rats-architecture.md [hunk 2]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+link is anchored to a Root of Trust from an entity that established the
+communications link
+or may have received evidence that the communications link is anchored to a
+Root of Trust from an entity that established the communications link.
+In the latter case, anchoring the communications link to a Root of Trust would mean
+that this entity only established a communications link with the attesting
+device after verifying that the device had a Root of Trust.
+
+As a result, the Verifier may be able to trust the attestation evidence from the device without
+an additional endorsement or even a cryptographically-verifiable signature of the
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:714]:
```suggestion
As a result, the Verifier may be able to trust the attestation evidence from the device without
endorsement or cryptographically-verifiable signature of the
```

File: draft-ietf-rats-architecture.md [hunk 3]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+Implicit trust can also be tied to the communications link over which
+the Evidence is conveyed. As an example, the Attester and Verifier can
+communicate over a link that is established using a Root of Trust.  An example of such communications
+is over a cellular link, where the link is established by the Attester leveraging an enclave (e.g., SIM card).  The Verifier may have established the communications link and verified
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:703]: Two sentences starting "As an example, ... An example of such ..." reads poorly.

```suggestion
communicate over a link that is established using a Root of Trust, such as a cellular link established by the Attester leveraging an enclave (e.g., SIM card).  The Verifier may have established the communications link and verified
```

File: draft-ietf-rats-architecture.md [hunk 4]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+the presence of the Root-of-Trust in doing so, or may have received evidence that the communications
+link is anchored to a Root of Trust from an entity that established the
+communications link
+or may have received evidence that the communications link is anchored to a
+Root of Trust from an entity that established the communications link.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:708]: Redundant with lines 704-706

File: draft-ietf-rats-architecture.md [hunk 5]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+In the latter case, anchoring the communications link to a Root of Trust would mean
+that this entity only established a communications link with the attesting
+device after verifying that the device had a Root of Trust.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:711]: I don't think this wording is correct.   Maybe "a Root of Trust that this entity trusts"?

File: draft-ietf-rats-architecture.md [hunk 6]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+
+As a result, the Verifier may be able to trust the attestation evidence from the device without
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:713]:
```suggestion
As a result, the Verifier may be able to trust the attestation Evidence from the Attester without
```

File: draft-ietf-rats-architecture.md (hunk 2 above)
Comment (dthaler) [reply draft-ietf-rats-architecture.md:714]: Maybe noun/verb mismatch?  I don't think a policy trusts something, I think the Verifier trusts something based on policy.

File: draft-ietf-rats-architecture.md [hunk 7]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+an additional endorsement or even a cryptographically-verifiable signature of the
+evidence.  Note that the appraisal policy employed by the Verifier may take into account the
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:715]:
```suggestion
Evidence.  Note that the Appraisal Policy employed by the Verifier may take into account the
```

File: draft-ietf-rats-architecture.md [hunk 8]
```diff
@@ -697,6 +697,25 @@ for by hardware or by ROM code, especially if such hardware is
...
+transient nature of the communications link security.  For instance, once the link is torn down
+then the Evidence may not be trusted until it is conveyed again over a re-established link.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:717]:
```suggestion
then the Evidence might not be trusted until it is conveyed again over a re-established link.
```

File: draft-ietf-rats-architecture.md (hunk 3 above)
Comment (nedmsmith) [reply draft-ietf-rats-architecture.md:703]: I can't tell if this paragraph is intended to add intuition to "Root of Trust" or to describe the duties of a Verifier that leverages a pre-existing comms session to protect the integrity of Evidence (and to ascribe the semantics of the key that the Attester 'claims' the assertions in the Evidence are accurate representation of current state of the Target environment).

File: draft-ietf-rats-architecture.md (hunk 5 above)
Comment (nedmsmith) [reply draft-ietf-rats-architecture.md:711]: It is at least verbose.

File: draft-ietf-rats-architecture.md (hunk 8 above)
Comment (nedmsmith) [reply draft-ietf-rats-architecture.md:717]: This whole section seems to be more that of a use case that describes an architectural principle or concept than directly stating the principle or concept.  

I think the principle is:
- A conveyance protocol that is authenticated and integrity protected can be used to convey unprotected Evidence IF the following properties hold:
a) The key used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment that collected claims about the Target Environment.
b) The Evidence in the conveyance channel is supplied by the Attesting Environment.
c) The Root of Trust protects the key and the Attesting Environment with equivalent strength.

File: draft-ietf-rats-architecture.md (hunk 1 above)
Comment (nedmsmith) [reply draft-ietf-rats-architecture.md:700]: +1

File: draft-ietf-rats-architecture.md (hunk 8 above)
Comment (dthaler) [reply draft-ietf-rats-architecture.md:717]: Agree and I like Ned's articulation above. Maybe the PR should just say what's in Ned's principle?

Comment (gmandyam) [reply draft-ietf-rats-architecture.md:717]: I have no objection.  It is concise and captures my original inten.  @nedmsmith - will you be taking the lead on the PR revision.

File: draft-ietf-rats-architecture.md [hunk 9]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+A conveyance protocol that is authenticated and integrity protected can be used
```
Comment (henkbirkholz) [inline draft-ietf-rats-architecture.md:700]: I assume what is meant here is:

* evidence is protected
* remote peers are (mutually) authenticated

File: draft-ietf-rats-architecture.md [hunk 10]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
...
+to convey unprotected Evidence, assuming the following properties exists:
+
+   1. The key used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment that collected claims about the Target Environment.
```
Comment (henkbirkholz) [inline draft-ietf-rats-architecture.md:703]: keys? of at least two remote peers? We are talking about a "Secure Channel" to convey unprotected Evidence right?

Also? In theory, unprotected Evidence are just Claims?

I am not sure if we should water down evidence, in general. Evidence should be tamper-evident. In consequent, messing with evidence conveyed via "secure channel" must be evident with a similar level of assurance.

File: metadata.min.js [hunk 11]
```diff
@@ -0,0 +1 @@
...var _=a[e][t].slice(0,3),v=a[e][t].slice(3).replace(/^0+/,"");L=C<y?"RFC"!=_?L+"<a href='"+s+"/info/"+_.toLowerCase().concat(v.toLowerCase())+"'>"+_+" "+v+"</a>, ":L+"<a href='"+s+"/info/"+_.toLowerCase().concat(v.toLowerCase())+"'>"+v+"</a>, ":"RFC"!=_?L+"<a href='"+s+"/info/"+_.toLowerCase().concat(v.toLowerCase())+"'>"+_+" "+v+"</a>":L+"<a href='"+s+"/info/"+_.toLowerCase().concat(v.toLowerCase())+"'>"+v+"</a>",C++}a[e]=L}else if("errata_url"==e){var R="";R=a[e]?R+"<a href='"+a[e]+"'>Errata exist</a> | <a href='"+d+"/"+l+"'>Datatracker</a>| <a href='"+n+"/?"+i+"="+f+"&submit="+i+"'>IPR</a> | <a href='"+c+"/"+l+"'>Info page</a>":"<a href='"+d+"/"+l+"'>Datatracker</a> | <a href='"+n+"/?"+i+"="+f+"&submit="+i+"'>IPR</a> | <a href='"+c+"/"+l+"'>Info page</a>",a[e]=R}""!=a[e]?"Errata"==u[e]?h+=`<dt>More info:</dt><dd>${a[e]}</dd>`:h+=`<dt>${u[e]}:</dt><dd>${a[e]}</dd>`:"Errata"==u[e]&&(h+=`<dt>More info:</dt><dd>${a[e]}</dd>`)}),h+="</dl>",e.innerHTML=h}catch(e){console.log(e)}else console.log("Could not locate metadata <div> element");function r(e){return e.charAt(0).toUpperCase()+e.slice(1)}}window.removeEventListener("load",addMetadata),window.addEventListener("load",addMetadata);
```
Comment (dthaler) [inline metadata.min.js:1]: What is this file for?

File: draft-ietf-rats-architecture.md (hunk 10 above)
Comment (mcr) [inline draft-ietf-rats-architecture.md:703]:
```suggestion
   1. The key used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment(s) that collected claims about the Target Environment(s).
```

File: draft-ietf-rats-architecture.md [hunk 12]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
...
+   2. The Evidence in the conveyance channel is supplied by the Attesting Environment.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:704]: same question about "the Attesting Environment"... which Attesting Environment?  Also the phrase reads as a tautology, can we wordsmith somehow?

Comment (mcr) [inline draft-ietf-rats-architecture.md:704]:
```suggestion
   2. All unprotected Evidence that is conveyed is supplied exclusively by the Attesting Environment that has the key that protects the conveyance channel
```

File: draft-ietf-rats-architecture.md (hunk 9 above)
Comment (mcr) [inline draft-ietf-rats-architecture.md:700]:
```suggestion
A conveyance protocol that provides authentication and integrity protection can be used
```

File: draft-ietf-rats-architecture.md [hunk 13]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+A conveyance protocol that provides authentication and integrity protection can be used
+to convey unprotected Evidence, assuming the following properties exists:
+
+   1. The key used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment(s) that collected claims about the Target Environment(s).
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:703]:
```suggestion
   1. The key material used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment(s) that collected claims about the Target Environment(s).
```

File: draft-ietf-rats-architecture.md [hunk 14]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+A conveyance protocol that provides authentication and integrity protection can be used
+to convey unprotected Evidence, assuming the following properties exists:
+
+   1. The key material used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment(s) that collected claims about the Target Environment(s).
+   2. All unprotected Evidence that is conveyed is supplied exclusively by the Attesting Environment that has the key that protects the conveyance channel
+   3. The Root of Trust protects both the conveyance channel key and the Attesting Environment with equivalent strength protections.
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:705]:
```suggestion
   3. The Root of Trust protects both the conveyance channel key material and the Attesting Environment with equivalent strength protections.
```

File: draft-ietf-rats-architecture.md [hunk 15]
```diff
@@ -697,6 +697,13 @@ for by hardware or by ROM code, especially if such hardware is
 physically resistant to hardware tampering.  The component that is
 implicitly trusted is often referred to as a Root of Trust.
 
+A conveyance protocol that provides authentication and integrity protection can be used
+to convey unprotected Evidence, assuming the following properties exists:
+
+   1. The key material used to authenticate and integrity protect the conveyance channel is trusted by the Verifier to speak for the Attesting Environment(s) that collected claims about the Target Environment(s).
+   2. All unprotected Evidence that is conveyed is supplied exclusively by the Attesting Environment that has the key that protects the conveyance channel
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:704]:
```suggestion
   2. All unprotected Evidence that is conveyed is supplied exclusively by the Attesting Environment that has the key  material that protects the conveyance channel
```

**Pull request merged**